subsurface/io/grids/surface_reader.py

```diff
@@ -0,0 +1,63 @@
+import pandas as pd
+from subsurface.structs.base_structures import UnstructuredData
+import numpy as np
+import math
+from scipy.spatial import Delaunay
+
+from subsurface.structs.errors import VertexMissingError
```

Comment:
In general, it is recommended not to create self defined error except if there is a very good reason for it. I think it is better to substitute this error by a KeyError or ValueError